openid-sharedsignals-eventdef-specification-1_0.md

```diff
@@ -0,0 +1,282 @@
+      },
+      "reason_admin":{
+         "type":"object",
+         "properties":{
+            "en":{
+               "type":"string"
+            }
+         }
+      },
+      "reason_user":{
+         "type":"object",
+         "properties":{
+            "en":{
+               "type":"string"
+            },
+            "es-410":{
+               "type":"string"
+            }
+         }
+      },
+      "event_timestamp":{
+         "type":"number"
+      }
+   }
+}
+~~~
+
+## Properties 
+
+The top level "properties" key contains an object where each property represents a key in the JSON data that’s being validated. Within each property, all normative requirements and non-normative details are specified by using the vocabulary in {{JSONSchemaValidation}}.
+
+The following is a non-normative example of a "properties" object for a CAEP event schema.
+~~~json
+{
+   "properties": {
+    "initiating_entity": {
+      "description": "Describes the entity that invoked the event.",
+      "type": "string",
+      "oneOf": [
+        {
+          "const": "admin",
+          "description": "an administrative action triggered the event"
+        },
+        {
+          "const": "user",
+          "description": "an end-user action triggered the event"
+        },
+        {
+          "const": "policy",
+          "description": "a policy evaluation triggered the event"
+        },
+        {
+          "const": "system",
+          "description": "a system or platform assertion triggered the event"
+        }
+      ]
+    },
+    "required": [ "initiating_entity" ]
+   }
+}
+~~~
+# Registry
+
+This section serves as a registry for the schemas of all registered SSF Event Types.
+
+| Event Type | Schema URI | Description
+|------|-------------|
+| CAEP  | <schema uri here> | brief description |
+| RISC  |  <schema uri here>  | brief description |
+
+{: title="Registered Event Types" #eventtypestable}
+
+## The Registration Process
+
+SSF Implementers may find that existing registered SSF event types do not meet the needs of their applications. In that case, they may propose a new SSF event type and register its schema. To do so, an implementer MUST create a request in the form of a Pull Request to https://github.com/openid/sharedsignals and meet the following requirements. The pull request will be reviewed by the Shared Signals Working Group and accepted at their discression.
+
+1. Author(s) of the pull request MUST be at least a contributing member of the OpenID Foundation.
+1. The Pull Request MUST contain a human readable description of the new SSF event type.
+1. The $id of the schema MUST be publicly accesbile on the internet and resolve to the schema document.
+1. The "title" and "description" of the schema must be meaningful and indicative of its function.
+1. The naming of all schema properties MUST be indicative of its function.
+1. Schemas may not be removed, only deprecated. Any changes to schemas must follow semantic versioning.
+
```

Comment:
Should we specify JSON schema for the SET and call out all the mandatory claims of the SET are required? This will help developers to generate entire container objects, rather than only the sub-object `events`